samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManagerRestoreUtil.java

```diff
@@ -338,7 +327,9 @@ private static BlobStoreManager getBlobStoreManager(Config config, ExecutorServi
     BlobStoreConfig blobStoreConfig = new BlobStoreConfig(config);
     String blobStoreManagerFactory = blobStoreConfig.getBlobStoreManagerFactory();
     BlobStoreManagerFactory factory = ReflectionUtil.getObj(blobStoreManagerFactory, BlobStoreManagerFactory.class);
-    return factory.getRestoreBlobStoreManager(config, executor);
+    BlobStoreManager blobStoreManager = factory.getRestoreBlobStoreManager(config, executor);
+    blobStoreManager.init();
```

Comment:
This method should not init this, it's caller should.

Where is this and BlobStoreBackupManager being closed? Should also be closed by the caller.

Reply:
Removed and added a cleanup step in the end to close this blobStoreManager.